service/src/main/kotlin/io/provenance/explorer/grpc/extensions/MsgConverter.kt

```diff
@@ -304,14 +310,34 @@ enum class IbcDenomEvents(val eventType: IbcEventType, val event: String, val id
     ACKNOWLEDGE_DENOM(IbcEventType.DENOM, "fungible_token_packet", "denom")
 }
 
-fun Any.getIbcEvents() =
+fun Any.getIbcDenomEvents() =
     when {
         typeUrl.contains("MsgRecvPacket") -> listOf(IbcDenomEvents.RECV_PACKET_DENOM, IbcDenomEvents.RECV_PACKET_ADDR)
         typeUrl.contains("MsgAcknowledgement") -> listOf(IbcDenomEvents.ACKNOWLEDGE_DENOM)
         else -> listOf<IbcDenomEvents>()
             .also { logger().debug("This typeUrl is not yet supported in as an ibc-event-based msg: $typeUrl") }
     }
 
+// Returns Pair(Event type, Pair(port, channel))
+fun Any.getIbcChannelEvents() =
+    when {
+        typeUrl.contains("MsgTransfer") -> "send_packet" to Pair("packet_src_port", "packet_src_channel")
+        typeUrl.contains("MsgChannelOpenInit") -> "channel_open_init" to Pair("port_id", "channel_id")
+        typeUrl.contains("MsgChannelOpenTry") -> "channel_open_try" to Pair("port_id", "channel_id")
+        typeUrl.contains("MsgChannelOpenAck") -> "channel_open_ack" to Pair("port_id", "channel_id")
+        typeUrl.contains("MsgChannelOpenConfirm") -> "channel_open_confirm" to Pair("port_id", "channel_id")
+        typeUrl.contains("MsgChannelCloseInit") -> "channel_close_init" to Pair("port_id", "channel_id")
+        typeUrl.contains("MsgChannelCloseConfirm") -> "channel_close_confirm" to Pair("port_id", "channel_id")
+        typeUrl.contains("MsgRecvPacket") -> "recv_packet" to Pair("packet_dst_port", "packet_dst_channel")
+        typeUrl.contains("MsgTimeout") -> "timeout_packet" to Pair("packet_src_port", "packet_src_channel")
+        typeUrl.contains("MsgAcknowledgement") -> "acknowledge_packet" to Pair("packet_src_port", "packet_src_channel")
+        else -> null.also { logger().debug("This typeUrl is not yet supported in as an ibc-event-based msg: $typeUrl") }
+    }
+
+// The only case where a channel is not in the events, because no events get emitted
```

Comment:
Are you thinking of ever putting the connection handshake events in this?

Reply:
they get recorded